### ISSUE TITLE
💅 False positive `noUselessEscapeInString` for `\${` in template literals (2.0.0-beta.3)

### ISSUE THREAD

**Issue by arumakan1727** [biomejs/biome#6039]
### Environment information

```bash
CLI:
  Version:                      2.0.0-beta.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.0.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.11.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    nursery/noUselessEscapeInString

Workspace:
  Open Documents:               0
```

### Rule name

lint/nursery/noUselessEscapeInString

### Playground link

https://next.biomejs.dev/playground/?lintRules=all&code=YABcACQAewB9AGAA

### Expected result

The backslash before the `${` in a template literal (e.g. <code>\`\${}\`</code> ) **should not** be flagged as a useless escape.

According to the [ECMAScript Lexical Grammar](https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-CharacterEscapeSequence), `\$` is treated as a *CharacterEscapeSequence* with a _**NonEscapeCharacter**_, and in most contexts, it is indeed considered a useless escape.

However, in template literals, the sequence `\${` has a special use: the backslash **prevents** the `${` sequence from being interpreted as the start of a substitution.
This means that <code>\`\${}\`</code> will render as `${}` in the resulting string, rather than interpolating an expression.

Therefore, the escape in this context is **not** useless; it is required to produce a literal `${` in the string output.

Note that when the backslash is not immediately followed by a `{`, such as `\$x`, the escape is indeed useless and I believe it should be flagged as a lint error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
